ra2ce/network/networks_utils.py

```diff
@@ -1539,6 +1539,43 @@ def fraction_flooded(line: LineString, hazard_map: str):
     except Exception as e:
         logging.info("fraction_flooded() {} \n for line {}".format(e, line))
 
+def fraction_flooded_array(line: LineString, raster_array: np.ndarray, affine) -> float:
+    """
+    Calculates the fraction of a linestring that overlaps with a hazard raster array (values > 0).
+
+    Args:
+        line (LineString): The linestring to check.
+        raster_array (np.ndarray): The raster data array.
+        affine: The affine transform for the raster.
+
+    Returns:
+        float: Fraction of the linestring overlapping raster cells with value > 0.
+    """
+    bbox_line = box(*line.bounds)
+    try:
+        # Mask the raster array with the bounding box of the line
+        out_image, out_transform = mask(
+            {"type": "raster", "data": raster_array, "transform": affine}, #instead of giving src as argument, give a dict with type, data and transform
+            [bbox_line], crop=True, all_touched=True
+        )
+
+        flooded_cells = unary_union(
+            [
+                shape(x[0])
+                for x in shapes(out_image[0], transform=out_transform)
+                if x[-1] > 0
+            ]
+        )
+        flooded_cells = gpd.GeoDataFrame({"geometry": [flooded_cells]})
+
+        line_intersect = flooded_cells.intersection(line)
+        return line_intersect.length.sum() / line.length
+    except ValueError:
+        return 0
+    except Exception as e:
```

Comment:
Given you are returning always the same value it does not make sense to have a specialized `ValueError` exception handling.